pkg/tls_extensions/atls_listener.c

```
@@ -0,0 +1,550 @@
+}
+
+int tls_close(tls_connection *conn) {
+    if (conn != NULL) {
+        
+        if (conn->ssl != NULL) {
+            int ret = SSL_shutdown(conn->ssl);
+
+            if (ret < 0) {
+                perror("SSL did not shutdown correctly");
+                free(conn);
+                close(conn->socket_fd);
```

cppcheck

[error] Dereferencing 'conn' after it is deallocated / released

(deallocuse)